synapse/events/third_party_rules.py

```diff
@@ -164,20 +183,22 @@ def register_third_party_rules_callbacks(
 
     async def check_event_allowed(
         self, event: EventBase, context: EventContext
-    ) -> Union[bool, dict]:
+    ) -> Tuple[bool, dict]:
         """Check if a provided event should be allowed in the given context.
 
         The module can return:
             * True: the event is allowed.
             * False: the event is not allowed, and should be rejected with M_FORBIDDEN.
-            * a dict: replacement event data.
+
+        If the event is allowed, the module can also return a dictionary to use as a
+        replacement for the event.
```

Comment:
```suggestion
        replacement for the event's content.
```
?

Reply:
No, it's the event, not just its content.